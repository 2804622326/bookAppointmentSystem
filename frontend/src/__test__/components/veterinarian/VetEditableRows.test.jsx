import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import VetEditableRows from '../../../components/veterinarian/VetEditableRows.jsx';
jest.mock('../../../components/veterinarian/VetSpecializationSelector.jsx', () => () => <div>selector</div>);

test('renders editable rows snapshot', () => {
  const vet = { id:1, firstName:'A', lastName:'B', email:'e', phoneNumber:'1', gender:'Male', specialization:'Surg' };
  const { asFragment } = render(<table><tbody><VetEditableRows vet={vet} onSave={()=>{}} onCancel={()=>{}} /></tbody></table>);
  expect(asFragment()).toMatchSnapshot();
});
