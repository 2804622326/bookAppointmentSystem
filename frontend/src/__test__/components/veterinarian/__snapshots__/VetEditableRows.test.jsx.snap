// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders editable rows snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td>
          <input
            class="form-control"
            name="firstName"
            type="text"
            value="A"
          />
        </td>
        <td>
          <input
            class="form-control"
            name="lastName"
            type="text"
            value="B"
          />
        </td>
        <td>
          <input
            class="form-control"
            name="email"
            type="email"
            value="e"
          />
        </td>
        <td>
          <input
            class="form-control"
            name="phoneNumber"
            type="text"
            value="1"
          />
        </td>
        <td>
          <select
            class="form-control"
            name="gender"
          >
            <option
              value=""
            >
              Select Gender
            </option>
            <option
              value="Male"
            >
              Male
            </option>
            <option
              value="Female"
            >
              Female
            </option>
          </select>
        </td>
        <td>
          <div>
            selector
          </div>
        </td>
        <td>
          <button
            class="me-2 btn btn-success btn-sm"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.97 4.97a.75.75 0 0 1 1.07 1.05l-3.99 4.99a.75.75 0 0 1-1.08.02L4.324 8.384a.75.75 0 1 1 1.06-1.06l2.094 2.093 3.473-4.425z"
              />
            </svg>
          </button>
          <button
            class="btn btn-secondary btn-sm"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708"
              />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
